File: cellab_devkit-1.0 (1)/rules_java/forest.java
/*

    Skeleton rule for Forest.  This is used only to set the
    world type and load the palette and evaluator.

    No lookup table is used.

*/

class Forest extends ruletable {
    void jcruleModes() {
        setWorld(13);                   // User evaluator, closed world
        setOwnCodeRequest("forest");    // User evaluator function
        setPatternRequest("zero");
        setPaletteRequest("forest");
    }

    int jcrule(int oldstate) {
        return 0;               // No lookup table
    }
}

public class forest {
    public static void main(String args[]) {
        (new Forest()).generateRuleFile("forest");
    }
}

File: cellab_devkit-1.0 (1)/ruledefs/forest.js
    /*

        Skeleton rule for Forest.  This is used only to set the
        world type and load the palette and evaluator.

        No lookup table is used.

    */

    rule.worldtype = 13;        // 2D closed world, user evaluator
    rule.ocodereq = "forest";
    rule.patreq = "zero";
    rule.palreq = "forest";

    function forest(oldstate) {
        return 0;               // No lookup table
    }
